core/src/main/java/com/scalar/db/storage/dynamo/DynamoAdmin.java

```diff
@@ -217,9 +222,34 @@ private ApplicationAutoScalingClient createApplicationAutoScalingClient(DynamoCo
   }
 
   @Override
-  public void createNamespace(String nonPrefixedNamespace, Map<String, String> options) {
-    // In Dynamo DB storage, namespace will be added to table name as prefix along with dot
-    // separator.
+  public void createNamespace(String nonPrefixedNamespace, Map<String, String> options)
+      throws ExecutionException {
+    Namespace namespace = Namespace.of(namespacePrefix, nonPrefixedNamespace);
+    try {
+      boolean noBackup = Boolean.parseBoolean(options.getOrDefault(NO_BACKUP, DEFAULT_NO_BACKUP));
+      createNamespacesTableIfNotExists(noBackup);
+      insertIntoNamespaceTable(namespace);
+    } catch (ExecutionException e) {
+      throw new ExecutionException("creating the namespace " + namespace + " failed", e);
+    }
+  }
+
+  private void insertIntoNamespaceTable(Namespace namespace) throws ExecutionException {
+    Map<String, AttributeValue> itemValues = new HashMap<>();
+    itemValues.put(NAMESPACES_ATTR_NAME, AttributeValue.builder().s(namespace.prefixed()).build());
+    try {
+      client.putItem(
+          PutItemRequest.builder()
+              .tableName(ScalarDbUtils.getFullTableName(metadataNamespace, NAMESPACES_TABLE))
+              .item(itemValues)
+              // Will cause to throw a ConditionalCheckFailedException if the namespace already
+              // exists
+              .conditionExpression("attribute_not_exists(" + NAMESPACES_ATTR_NAME + ")")
+              .build());
+    } catch (ConditionalCheckFailedException e) {
+      throw new ExecutionException(
+          "the namespace " + namespace + " cannot be created as it already exists");
+    }
```

Comment:
We should also handle other runtime exceptions here?
```suggestion
    } catch (Exception e) {
       ...
    }
```

Reply:
Thank you. I revised it, please have a look again.